### PR TITLE
Base identity platform - initial add

### PR DESCRIPTION
**Pull request by darzu**
Adds a identity experiment which enables a new "login" button on the homepage which pops a dialog with a few login options.

### CODE REVIEW

**Review by eanders-MS — APPROVED** [review]
LGTM, but does this need to be merged to master? Seems early.

File: pxtlib/emitter/cloud.ts
Comment (darzu) [inline pxtlib/emitter/cloud.ts:6]: Actually, this might not work with staging.pxt.io
I'll need to rethink this

**Comment by darzu** [comment]
> LGTM, but does this need to be merged to master? Seems early.

Merging into master is easiest since arcade staging will point to that naturally. Also that way I'm not slowing down others that want to test things in staging.

The thought is that since it'll be behind an experiment it should be out of the way, and it gives us a hook to try other identity stuff.

File: pxtlib/emitter/cloud.ts
Comment (eanders-MS) [reply pxtlib/emitter/cloud.ts:6]: Does the client app have a per-environment config file?

Comment (darzu) [reply pxtlib/emitter/cloud.ts:6]: @eanders-MS , can you think of any reason why we wouldn't use "window.location.origin" here ?

Comment (eanders-MS) [reply pxtlib/emitter/cloud.ts]: That should work, but might need to special case it for localhost to fixup the port number.

File: webapp/src/identity.tsx
Comment (darzu) [inline webapp/src/identity.tsx:16]: extra query param flag makes it harder to stumble upon

File: pxtlib/emitter/cloud.ts
Comment (darzu) [reply pxtlib/emitter/cloud.ts]: I decided against this change since "baseEndpoint" is only used when running in node or localhost, otherwise we use site relative "/api" as the api root which is effectively like using window.location.origin

**Comment by darzu** [comment]
Most of this was reviewed here: https://github.com/microsoft/pxt/pull/7506#event-3868305713

**Comment by darzu** [comment]
@eanders-MS , think this is ready for merge?

**Comment by eanders-MS** [comment]
@darzu, we should hold off merging until we have privacy & security approval.

**Comment by eanders-MS** [comment]
@darzu, there's also the CORS thing. I think I have a solution for this now.

**Comment by eanders-MS** [comment]
@darzu, I also want to get session tokens in place before this gets merged to master.

**Review by abchatra — APPROVED** [review]

**Review by eanders-MS — APPROVED** [review]
:shipit:

**Pull request merged**
